docs/running-locally.md

```diff
@@ -1,23 +1,51 @@
-- You will need to install the node depdendencies. You only need this step once
-
+# Run cardanow locally
+## Setup tooling
+In order to work with the project scripts some tools are (some examples are `cardano-cli` and `mithril-client`). If you run nix, everything will be available in the nix flakes shell, these commands will enable a shell with all the tools that will be in this guide:
 ```
-npm i
+git submodule update --init --recursive
+nix develop
 ```
 
-- Use the `dowload-mithril-snapshot.sh` script to download the snapshot. You will need to specify a NETWORK env var
+If you are not a nix user, you will have to manually install the various tools:
+- docker
+- docker-compose
+- cardano-cli
+- mithril-client-cli
+- jq
+- nodejs
 
-```
-NETWORK=preview ./components/cardano-node/dowload-mithril-snapshot.sh
-```
+**NOTE**: the versions of these tools are not yet bounded, the use case is still small enough that no version constrains are required yet, in the future che versions of at least the `mithril-client` and the `cardano-cli` will be specified.
 
-- Start the compose environment. You will need to specify a NETWORK env var
+## Install nodejs dependencies
+You will need to install the node dependencies. You only need this step once
 
+```shell
+npm i
 ```
-NETWORK=preview docker-compose up -d
+
+## Download Mithril snapshot
+Use the `download-mithril-snapshot.sh` script to download the snapshot. You will need to specify a `NETWORK` env var
+
+```shell
+NETWORK=preview ./components/cardano-node/download-mithril-snapshot.sh
 ```
 
-- Run the application
+## Build environment - isolated network
+Start the compose environment. You will need to specify a `NETWORK` env var:
+```shell
+NETWORK=preview
+MITHRIL_CONFIG=configurations/mithril-configs/$NETWORK.env
+docker compose --env-file $MITHRIL_CONFIG up -d
+```
 
+To verify the cardano node is actually running you can run:
+```shell
+cardano-cli query tip --testnet-magic 2 --socket-path "$(docker volume inspect cardanow_cardano-node-ipc | jq -r '.[0]["Mountpoint"]')/node.socket"
```

Comment:
@nazrhom I know this is not very clean - the volumes are not exported so this is the fastest solution that came to my mind, we may want to change/remove this in the future (or now)

Reply:
This is actually a bit problematic. I remembered this does not work on macOS, because it is not possible to share a socket between host and containers :( (I opened an issue about this almost a year ago, but never really got any response https://github.com/IntersectMBO/cardano-node/issues/4913)
Maybe the best solution is to run cardano-cli from the node image directly, and avoid having it on the host? The reason we needed it in the first place, is that we need to figure out the latest block the node is aware of, maybe we could also fetch that info some other way??

Reply:
> maybe the best solution is to run cardano-cli from the node image directly

mh yes, in this case I'll move towards that direction :+1: